clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17172,6 +17179,113 @@ static void WriteCharValueForDiagnostic(uint32_t Value, const BuiltinType *BTy,
   }
 }
 
+/// A raw_ostream that clips its output beyond a certain size, inserting a
```

Comment:
I this we should move that to llvm/Support - either in raw_ostream.h or a new file